Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -86,7 +86,11 @@ struct BinaryDecoder: Decoder {
         fieldEndP = nil
         let start = p
         let c0 = start[0]
-        fieldWireFormat = c0 & 7
+        if let wireFormat = WireFormat(rawValue: c0 & 7) {
```

Comment:
We have `FieldTag` to isolate this bit twiddling away from the decoders themselves, but we don't have a method specifically for this case since you're working with only the least-significant byte of the tag (which I guess isn't a problem in itself, we can ignore the higher ones for this particular operation, it's just another reason this extremely low-level rewrite bums me out).

It would be worth checking to see if the compiler can optimize `if let wireFormat = FieldTag(rawValue: Int32(c0))?.wireFormat` enough that it boils down to the same operations you have here, at least.